src/Orleans.Core/Serialization/SerializationManager.cs

```diff
@@ -575,6 +594,19 @@ internal static void GetSerializationMethods(Type type, out MethodInfo copier, o
             }
         }
 
+        private Delegate CreateDelegate(MethodInfo methodInfo, Type delegateType, object target)
```

Comment:
delegateType could be a generic type parameter, to avoid all the unnecessary casting on the caller side

Reply:
I previously tried to do that, it wasn't possible to use a generic parameter and cast it in the return statement. It seems that if I use the `as` operator it will do the trick.

https://stackoverflow.com/questions/35321586/cannot-cast-delegate-to-a-generic-type-t